compiler/qsc_passes/src/baseprofck.rs

```diff
@@ -60,27 +57,6 @@ pub fn check_base_profile_compliance(package: &Package) -> Vec<Error> {
     checker.errors
```

Comment:
The `check_base_profile_compliance_for_stmt` function used to have this extra check to make sure that the statment being used to trigger QIR codegen didn't return anything that was incompatible with base profile, but now nothing checks that. It can caus a panic in codegen from Python when the provided expr returns something invalid:
```python
import qsharp
qsharp.init(qsharp.TargetProfile.Base)
qir = qsharp.compile("1")
```
Used to return `Qsc.BaseProfCk.ReturnNonResult` and now hits `PanicException: unexpected value type: Int(1)`. I think you might be able to handle this by checking the last top-level statement under the assumption that QIR codegen would be using that as the entry statement, which I think holds, but is not quit obvious. Something like this:
```rust
    if let Some(StmtKind::Expr(expr)) = &package.stmts.last().map(|stmt| &stmt.kind) {
        if any_non_result_ty(&expr.ty) {
            checker.errors.push(Error::ReturnNonResult(expr.span));
        }
    }
```
added into the new unified `check_base_profile_compliance` after the similar check on the entry expression.

Reply:
for what it's worth, I confirmed the above works to resolve the issue. Still worth thinking it through (and adding a test case).